thrift-parser.js

```diff
@@ -366,6 +410,34 @@ module.exports = (buffer, offset = 0) => {
     return result;
   };
 
+  const readUnion = () => {
+    let subject = readKeyword('union');
+    let name = readName();
+    let items = readUnionBlock();
```

Comment:
@lujjjh very interesting.  It also looks like multiple defaults is invalid syntax (and throws), so it seems the `readUnion` logic should remain separate (I'll have to add the singular default logic in another commit).  @YanagiEiichi what do you think?  Should we force each "option" value to "optional"?